sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/view/ViewOptionsImpl.scala

```diff
@@ -26,5 +26,5 @@ import java.util
  */
 private[akkaserverless] final case class ViewOptionsImpl(override val forwardHeaders: java.util.Set[String])
     extends ViewOptions {
-  def withForwardHeaders(headers: util.Set[String]): ComponentOptions = copy(forwardHeaders = headers)
+  def withForwardHeaders(headers: util.Set[String]): ViewOptions = copy(forwardHeaders = headers)
```

Comment:
Here we type the return to be the current impl. I run into a situation in an adapter that required some specialised type, but the method was returns ComponentOptions.